### ISSUE TITLE
Start PyLadies Member Database

### ISSUE THREAD

**Issue by lorenanicole** [pyladies/global-organizing#54]
# Overview

Now that #50 is open, per the timeline we have to start preparing for outreach with our PyLadies members to in April / May 2020 either vote or nominate themselves for the inaugural PyLadies Global Council. To do this, we need a distribution list to communicate with our PyLadies members directly.

## Deliverables

- [X] Define workflow for collecting members info (@lorenanicole) by Feb 2020 PyLadies Global Meeting
- [x] Week of ~Jan 23rd~ Feb 12th checkin (@deboraazevedo @lorenanicole) re: translation + outreach strategy plan 
- [x] Define outreach strategy 
      - Mediums
      - Assets for translation
      - How are we tracking outreach?
      - The Project Comms team is starting to help see: https://github.com/pyladies/project-communications/issues/24
- [X] Create list of questions:   
   - [X] [Basic Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/basic-membership-form-en.md) (english)
   - [X] [Voting Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/voting-membership-form-en.md) (english) 
- [ ] Translation (@deboraazevedo @elainewong can coordinate with @Mariatta)
     - [x] Portuguese - [Basic Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/basic-membership-form-pt.md) - [Voting Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/voting-membership-form-pt.md) - TY @deboraazevedo 
     - [x] Spanish - [Basic Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/basic-membership-form-es.md) - [Voting Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/voting-membership-form-es.md) TY https://github.com/alphacentauri82
     - [x] Mandarin - [Basic Membership Form](https://docs.google.com/document/d/12TTQxO5ySwQgT8E3OuQRZACkHDTv7Fanu6Ptj8qWeik/edit?usp=sharing) - [Voting Membership Form](https://docs.google.com/document/d/1yeGZu7tIr9ZJptUN32lpfaLcfLXceH_5KWk3UgAwA-o/edit?usp=sharing)
     - [ ] Arabic - [Basic Membership Form](https://docs.google.com/document/d/1Q3xIlcjXKiyb-u45zZkTGZrQso1YXf3zjljCmuWQpWs/edit?usp=sharing) - [Voting Membership Form](https://docs.google.com/document/d/1d7N-GY7tT6UFZW34CS55yWUaiLXci0NQhALvFWsPITg/edit?usp=sharing)
     - [X] English - [Basic Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/basic-membership-form-en.md) - [Voting Membership Form](https://github.com/pyladies/project-admin-logisitics/blob/master/forms/voting-membership-form-en.md)
     - [ ] French - [Basic Membership Form](https://docs.google.com/document/d/1YG66IfjBJGVPiElC2HMP9oKRb-LoWJDxdwNvZRlxNlQ/edit?usp=sharing) - [Voting Membership Form](https://docs.google.com/document/d/1MJooc4tji3hzAnt1FWxmUky3i9uuIDB51NU6_gSre0M/edit?usp=sharing)
     - [x] Russian - [Basic Membership Form](https://docs.google.com/document/d/11scrf8sOU6zoxnrQ_MXxdvBXdFQiAtDIFgl_vrU89PI/edit?usp=sharing) - [Voting Membership Form](https://docs.google.com/document/d/15vIcxJE0GpkZR-P-9kpp-FWz3Bjhl1crtdNMLyPB7g8/edit?usp=sharing)
- [x] Make form for collecting members info including an explicit consent to contact them regarding PyLadies business (@BethanyG) 
- [ ] Start registration by Mar 1 2020

## PyLadies Membership Registration

Proposal is for:
- Basic registration (no voting) 
- Voting registration that will include more demographic info 

See: https://github.com/pyladies/global-organizing/issues/54#issuecomment-583529133

*Per the [Feb 2020 PyLadies Global Meeting](https://github.com/pyladies/global-organizing/blob/master/notes/2020/13_02_2020.md) we decided the following #60*

**Registering for a chapter:** 

- All active chapters from pyladies directory will be a drop down in form
- Inactive chapters can be typed in per an `other` field
- Remote will be added
- "Not a local chapter" as well

**How will we disseminate registration**

- Through PyLadies organizers
- Through social media

**Verifying Members**

PyLadies chapter organizers will be able to "verify" their own members: - - No one external will weigh in
- PyLadies organizers will be provided a list with members
- We will use an 'opt out' model, unless an organizer 'opts out' a member (don't verify)

PyLadies with no local chapter, inactive, or remote:
- An additional registration step will be required
- We will take time to brainstorm and revisit at PyLadies Mar 2020 meeting #62 

## Questions

1. Where will member information live?

To start, in the PyLadies Google account (e.g. spreadsheets). We can restrict access only then to official `@pyladies.com` Google accounts.

The plan is to forward this data to the PSF when they move to a new CRM in 2020 per a chat with @ejodlowska. See: https://github.com/pyladies/global-organizing/issues/54#issuecomment-583013574.

2. Will the member information of a given chapter be shared with the chapter organizers? 

Yes. Per question 1, we will house the information initially within the PyLadies Google account. We can then restrict information using the `chapter@pyladies.com` Google accounts. See: https://github.com/pyladies/global-organizing/issues/54#issuecomment-583534276.

**Comment by trallard** [comment]
- As per consent, I'd suggest we make this GDPR compliant by default, this will avoid a lot of headaches
 when dealing with data usage/deletion
- About the form: we might need to explore something other than Google forms (which is very commonly used) as there are countries where these are blocked

**Comment by lorenanicole** [comment]
@trallard definitely 👍 on GDPR compliant. It was a total memory lapse when the initial form was made for Chapter Directory. 

As per the Member Directory what alternative to Google Forms do you suggest? I talked with the PSF and that’s what they have in place (a new CRM is coming sometime next year). I don’t want us locked into paid software if it isn’t necessary as we should be considering plugging into the PSF infra in the future. Specifically the thought is when people register as a PSF member one option for membership is as a PyLadies member!

**Comment by trallard** [comment]
I am not sure what alternatives would be suitable in this case. I think Typeform is generally a good alternative. But I will make sure to do more digging around other alternatives that won't prevent us from joining the PSF infra in the future.

**Comment by lorenanicole** [comment]
Typeform works for me! Thanks @trallard !

**Comment by lorenanicole** [comment]
# Questions

For the member chapter directory here's some proposed questions I think we should collect ...

* Name
* Pronouns 
* Email
* Slack Handle
* Professional title (dropdown with options like developer, student, academia/science,  data scientist, analyst, engineer, etc ..)
* PyLadies Chapter Affiliation
* Are you a chapter organizer?
* How do you use Python?
* What do you wish to see from PyLadies?
* Would you like to be more involved in PyLadies? If yes, do any of these project teams interest you?
* Are you a Python Software Foundation member? If yes, what level? 
* An "I consent to share my email" verbiage for GDPR compliance

**Comment by trallard** [comment]
That looks good. I think at least the following questions should allow for folks to self define (i.e. provide an "other": option):
- Pronouns
- Professional title

**Comment by jesford** [comment]
Some of these questions look like they would take free form responses (How do you use Python?
What do you wish to see from PyLadies?). Is this data useful to collect & store? Definitely seems interesting to have...

Also, if we're just trying to learn more about members, should we also ask:
- Have you presented at PyLadies?
- How long have you been using Python?

**Comment by AbigailMesrenyameDogbe** [comment]
Asking for members consent to receiving newsletters can be considered.
Also, is it useful to ask for Github handles? (could be an optional field).

**Comment by lorenanicole** [comment]
Updated with additional questions ...

# Questions

For the member chapter directory here's some proposed questions I think we should collect ...

** Member Information **

* Name
* Pronouns 
* Email
* Slack Handle
* Professional title (dropdown with options like developer, student, academia/science,  data scientist, analyst, engineer, etc ..)
* Language(s) spoken

** PyLadies Chapter Membership **

* PyLadies Chapter Affiliation
* Are you a chapter organizer?
* Have you presented at PyLadies?

** PyLadies Programming and Communication **

* What do you wish to see from PyLadies?
* Would you like to receive the PyLadies email newsletter? Expect emails quarterly.

** PyLadies Volunteering **

* Would you like to be more involved in PyLadies? If yes, do any of these project teams interest you: project communications, project admin & logistics ... ?

** Python Background ** 

* How long have you used Python?
* How do you use Python?
* Are you a Python Software Foundation member? If yes, what level? 

**  General GDPR Compliance ** 
* An "I consent to share my email" verbiage for GDPR compliance

**Comment by Mariatta** [comment]
There are a lot of questions here... Some thoughts:

- What is the purpose of the member database? I was under the impression that this is for the election, to find out who should receive the ballot.

- It seems to me, the questions under "PyLadies Programming and communication", "PyLadies volunteering", and "python backgroud" are things that could change year by year for each individuals. For example, they might be able to volunteer more this year, but not so much next year. So it seems more like "member preference" instead of "member stats/directory"? I think that we will need to provide a way for them to update the values under "programming", "volunteering", "python background" Other info like name, email address, slack handle, probably don't change very often, and it is perhaps ok to only capture these info one time in a form.

**Comment by Mariatta** [comment]
I understand we do want to collect these info, but it feels that these questions about "programming", "volunteering", "python background" are more like a survey type of question, rather than for building a member directory.

**Comment by lorenanicole** [comment]
@Mariatta I can definitely understand. 

That said here's some thoughts:

* What are we collecting this info for?

Great question! I think we need to maybe refocus. There's two options:
- We collect only for the election 
- We collect for a member database

* Why would we want to collect for a member database?

Outreach is hard, as we have seen with the PyLadies Chapter Directory. It takes _a while_. Making this a "two for one" effort means we can start collecting data points that are useful for Project teams (and perhaps for us to start developing internal heuristics for measuring PyLadies community and community engagement!).

* Doesn't collecting this info become static?

Yes, @Mariatta definitely we would want to check in on this. We could email members at the start of every year and confirm:

- Are you still a member? 
- Do you still want emails?
- Can you update your fields here?

TLDR I think it behooves us to start getting more info at the start but specifically start collecting now since we will start using this for the election. If we are concerned about maintenance or this being too much we can make everything but member info, chapter info, and waiver to say we can email optional.

Thoughts?

**Comment by jesford** [comment]
I think those are really good points @Mariatta. Maybe we should keep the data collection more minimal for the database purpose and break the rest into a survey if we are interested to know about about the current preferences and status of each member (or collect the data the same way but only store limited entries in the DB).

I think @lorenanicole 's idea of having members update transient data every year could work, but will require more effort than keeping a more basic list that people just subscribe and unsubscribe from.

**Comment by lorenanicole** [comment]
@jesford @Mariatta I asked the @pyladies/communications for a point person to help us develop a comms strategy for this.

Perhaps what we can do is take the questions and break them up to work into that strategy including:

- Registration (minimal, must have info) OR Registration and start with this info (I'm actually a big fan of this, we are already going to be doing the outreach ..)
- Annual Survey (used for comms outreach, and further project development)

**Comment by pfctdayelise** [comment]
As background, would someone familiar with it be able to summarise how membership/voting eligibility is handled in PSF? is there infra or experience there we could leverage?

**Comment by lorenanicole** [comment]
@pfctdayelise great question! I'm a Director with the Python Software Foundation and have raised this question in the on-site PSF Board meeting Nov 2019, and the answer right now is no, not right now.

The PSF staff is investigating moving to a new CRM but that won't be until after PyCon US 2020 (which the PSF staff runs). This should be an action item however that the new Council takes on when the PSF staff is ready (I can help liaison with this) in mid 2020.

That said, I've already spoken with @ejodlowska / PSF staff and we've all agreed that something like this would be a hypothetical future for when the PSF has new system in place:

*Registering*
- When registering to be a PSF member option an individual can select "PyLadies" as a qualifying member rationale (in addition to the other membership options). 

*Porting data*
- We can pass off data we collect to the new PSF CRM and house our data membership info there. We should be careful when this happens to consider privacy considerations and ensure we are GDPR, for example, compliant in doing this.

Does that answer your question?

**Comment by pfctdayelise** [comment]
> Does that answer your question?

Thanks, yes it does about the infra. What about the eligibility to vote? Seems like in PSF 'Basic' (self sign up) members can't vote, and you need to either pay an annual donation (membership fee) or apply as a managing/contributing member which is reviewed by the board. What's our equivalent expectation?

**Comment by lorenanicole** [comment]
@pfctdayelise I believe you are asking, "do all PyLadies registered members get to vote?". In which case, this is a great question and is something that needs to be discussed now that the PyLadies organizers have voted on using public election as part of the selection process. I believe that is beyond the scope of this issue. In fact, it is an issue on my list to open!

**Comment by lorenanicole** [comment]
We've got to start getting this moving ... and I actually just noticed a model that maybe is a good one for us to follow to start - the Code for America Bridge National Advisory Council elections. See: https://brigade.codeforamerica.org/about/national-advisory-council/2020-candidates.

To vote, members have to confirm they've taken the "census survey", which in effect means confirming membership in a brigade chapter. The survey then collects demographic info, which is a ls a proposal we have here.

That said, I am going to revisit my proposal: 


> @Mariatta I can definitely understand.
> 
> That said here's some thoughts:
> 
> _What are we collecting this info for?_
> 
> Great question! I think we need to maybe refocus. There's two options:
> 
>     * We collect only for the election
> 
>     * We collect for a member database
> 
> 
> _Why would we want to collect for a member database?_
> 
> Outreach is hard, as we have seen with the PyLadies Chapter Directory. It takes _a while_. Making this a "two for one" effort means we can start collecting data points that are useful for Project teams (and perhaps for us to start developing internal heuristics for measuring PyLadies community and community engagement!).
> 
> _Doesn't collecting this info become static?_
> 
> Yes, @Mariatta definitely we would want to check in on this. We could email members at the start of every year and confirm:
> 
>     * Are you still a member?
> 
>     * Do you still want emails?
> 
>     * Can you update your fields here?
> 
> 
> ## TLDR I think it behooves us to start getting more info at the start but specifically start collecting now since we will start using this for the election.
> 
> If we are concerned about maintenance or this being too much we can make everything but member info, chapter info, and waiver to say we can email optional.
> 
> Thoughts?

And suggest we setup the form as follows:

- Form is distributed by organizers via their own mechanism (e.g. email through MeetUp to registered members, at in person events)
- Form allows individuals to: register to be a member to vote, opt in to provide the additional info

**Comment by jesford** [comment]
This sounds great to me @lorenanicole! On point 3 we could require or request that members without a local chapter join PyLadies Remote (I just checked their site though and it doesn't look like they've been active lately, so maybe not).

**Comment by lorenanicole** [comment]
@jesford we can have them do that! I know @ossanna16 use to run that, @ossanna16 is anyone actively maintaining that? If not, could we reset that to perhaps be the global group for now so folks have a chapter they can run? I actually wouldn't mind being a "lead" in that I can schedule a remote chat more easily now these days than an IRL experience.

**Comment by ossanna16** [comment]
Hiya! I actually don't lead PyLadies Remote anymore. Please reach out to Corryn Smith about this. I can provide an email address in a private channel.

**Comment by trallard** [comment]
I might be missing something obvious, but where is this database going to live and who will have access to this? Is this going to be access by the PSF and interim gov team?

**Comment by lorenanicole** [comment]
@trallard the plan is to turn the database over to the PSF to house in their system when a new CMS is selected in later 2020. For now, we can house the information as a Google spreadsheet and restrict access to only those that are admins on the PyLadies Google account.

**Comment by lorenanicole** [comment]
# Registration for Basic Member (No Voting):

## Member Information

    Name
    Pronouns
    Email
    PyLadies Slack Handle
    GitHub Handle (optional)
    Social Media Handle (optional)
    Language(s) spoken

## PyLadies Chapter Membership

    PyLadies Chapter Affiliation

## PyLadies Membership Option
    
    Yes - I would like voting privileges. If yes, I will complete the extended membership voting form. 
    No - I would like basic membership. This entitles me to updates from PyLadies by email. 

## General GDPR Compliance

    An "I consent to share my email" verbiage for GDPR compliance

# Registration for Voting Member:


## Member Information
    Name
    Pronouns
    Email
    PyLadies Slack Handle
    GitHub Handle (optional)
    Social Media Handle (optional)
    Language(s) spoken
    Professional title (dropdown with options like developer, student, academia/science, data scientist, analyst, engineer, etc ..)
    Industry (dropdown with options like software, academia, marketing, etc ...)

## PyLadies Programming and Communication
    PyLadies Chapter Affiliation
    What programming would you like to see PyLadies invest in?
    What types of events do you find value in: networking, workshops, code and coffee / informal coding meetups, conferences, other: blank?
    Would you like to receive the PyLadies email newsletter? Expect emails quarterly.

## PyLadies Volunteering

    Would you like to be more involved in PyLadies? If yes, do any of these project teams interest you: project communications, project admin & logistics ... ?

## Python Background

    How long have you used Python?
    How do you use Python?
    Are you a Python Software Foundation member? If yes, what level?

## General GDPR Compliance

    An "I consent to share my email" verbiage for GDPR compliance

**Comment by wgrus** [comment]
Will the information be shared with the local organizers?

**Comment by lorenanicole** [comment]
@wgrus great question! I have amended the question section in this issue above to answer in question 2.

**Comment by Mariatta** [comment]
Not sure if this would be a lot of work, or is it possible at all: would be great if the "chapter affiliation" is a dropdown of known registered pyladies chapter, instead of a freeform text.

The rest looks good to me. Thanks.

**Comment by lorenanicole** [comment]
@Mariatta we will figure out a way! If we have to we can link to an external list and have them pick from there. 

> Not sure if this would be a lot of work, or is it possible at all: would be great if the "chapter affiliation" is a dropdown of known registered pyladies chapter, instead of a freeform text.

**Comment by lorenanicole** [comment]
We have extended the proposal for: registration for chapters, verification of members, and more in the proposal above. Please weigh in! We are aiming to start member registration Mar 1 2020.

**Comment by lorenanicole** [comment]
@deboraazevedo @elainewong here's the questions we need translated: https://gist.github.com/lorenanicole/73a142b1cee06cba7c6eed3dd237129e.

**Comment by treasurechristain** [comment]
> Updated with additional questions ...
> 
> # Questions
> For the member chapter directory here's some proposed questions I think we should collect ...
> 
> _Member Information_
> 
> * Name
> * Pronouns
> * Email
> * PyLadies Slack Handle
> * GitHub Handle  (optional)
> * Social Media Handle (optional)
> * Language(s) spoken
> * Professional title (dropdown with options like developer, student, academia/science,  data scientist, analyst, engineer, etc ..)
> * Industry (dropdown with options like software, academia, marketing, etc ...)
> 
> _PyLadies Chapter Membership_
> 
> * PyLadies Chapter Affiliation
> * Are you a chapter organizer? If yes, which chapter?
> * What was the most recent event you attended? Please link to the event page.
> * How long have you been involved in your PyLadies chapter?
> * Have you presented at PyLadies?
> 
> _PyLadies Programming and Communication_
> 
> * What programming would you like to see PyLadies invest in?
> * What types of events do you find value in: networking, workshops, code and coffee / informal coding meetups, conferences, other: blank?
> * Would you like to receive the PyLadies email newsletter? Expect emails quarterly.
> 
> _PyLadies Volunteering_
> 
> * Would you like to be more involved in PyLadies? If yes, do any of these project teams interest you: project communications, project admin & logistics ... ?
> 
> _Python Background_
> 
> * How long have you used Python?
> * How do you use Python?
> * Are you a Python Software Foundation member? If yes, what level?
> 
> _General GDPR Compliance_
> 
> * An "I consent to share my email" verbiage for GDPR compliance

I suggest that social media handle shouldn't be optional... At least one should be compulsory... Because it will help in networking with the members via other platforms.

**Comment by lorenanicole** [comment]
Thanks for the input @treasurechristain! One thing about requiring a social media handle, though, is not everyone uses social media. So unfortunately that’ll mean we need to keep it optional. 

Was there anything else?

**Comment by lorenanicole** [comment]
Any updates on the member registartion translation @deboraazevedo @elainewong ?

**Comment by deboraazevedo** [comment]
There was no edit requests on the docs, as Mariatta pointed out. I'm doing the Portuguese translation myself. Today Mariatta suggested to post this request on the general channel on Slack

**Comment by Mariatta** [comment]
I've posted in psf-community mailing list https://mail.python.org/pipermail/psf-community/2020-February/000770.html

**Comment by trallard** [comment]
I can take the Spanish translation. Sorry if I have been MIA

**Comment by Mariatta** [comment]
@trallard Spanish translation had started [here](https://github.com/pyladies/project-admin-logisitics/pull/31)

Perhaps you can help review it? Thanks!

**Comment by lorenanicole** [comment]
@Mariatta did you mean https://github.com/alphacentauri82 has done the translation?

**Comment by Mariatta** [comment]
Yes they are at https://github.com/pyladies/project-admin-logisitics/blob/master/forms/basic-membership-form-es.md
and https://github.com/pyladies/project-admin-logisitics/blob/master/forms/voting-membership-form-es.md

**Comment by lorenanicole** [comment]
Updated the issue already!

**Comment by treasurechristain** [comment]
Nice work so far...

**Comment by lorenanicole** [comment]
This is already done ! CLOSING!
